pkg/resource/deploy/source_eval.go

```diff
@@ -850,6 +849,9 @@ func (rm *resmon) ReadResource(ctx context.Context,
 		ref, provErr := rm.defaultProviders.getDefaultProviderRef(providerReq)
 		if provErr != nil {
 			return nil, provErr
+		} else if providers.IsDenyDefaultsProvider(ref) {
+			msg := diag.GetDefaultProviderDenied("Read").Message
```

Comment:
I believe this function takes a urn as argument, do we have one here?

Reply:
The urn is generated by the provider after the read. So we don't have access to it yet.